sct.py

```diff
@@ -487,85 +488,82 @@ def list_resources(ctx, user, test_id, get_all, get_all_running, verbose):
         click.secho("Nothing found for selected filters in Docker!", fg="yellow")
 
 
-@cli.command('list-ami-versions', help='list Amazon Scylla formal AMI versions')
-@click.option('-r', '--region',
-              type=CloudRegion(cloud_provider="aws"),
-              default='eu-west-1',
-              help="a region to look for AMIs (default: eu-west-1)")
-@click.option('-a', '--arch',
-              type=click.Choice(AwsArchType.__args__),
-              default='x86_64',
-              help="architecture of the AMI (default: x86_64)")
-def list_ami_versions(region: str, arch: AwsArchType):
-    add_file_logger()
-
-    tbl = PrettyTable(field_names=["Name", "ImageId", "CreationDate"], align="l")
-    for ami in get_scylla_ami_versions(region_name=region, arch=arch):
-        tbl.add_row([ami.name, ami.image_id, ami.creation_date])
-    click.echo(tbl.get_string(title="Scylla AMI versions"))
-
-
-@cli.command('list-ami-branch', help="""list Amazon Scylla branched AMI versions
-    \n\n[VERSION] is a branch version to look for, ex. 'branch-2019.1:latest', 'branch-3.1:all'""")
+@cli.command('list-images', help="List machine images")
+@click.option('-b', '--backend',
+              type=click.Choice(["aws", "gce", "azure"]),
+              help="Cloud backend to query images for.")
+@click.option('-br', '--branch',
+              type=str,
+              default="master:latest",
+              help="Branch to query images for. Defaults to 'master:latest'")
+@click.option('-v', '--version',
+              type=str,
+              help="List images by version. Use '-v all' for all versions. "
+                   "OSS format: <4.3> Enterprise format: <enterprise-2021.1>")
 @click.option('-r', '--region',
-              type=CloudRegion(cloud_provider="aws"),
```

Comment:
Why didn't you kept the CloudRegion ? It can validate the input of region based on the backend

Reply:
We need to provide the `cloud_provider` argument in the signature though. IIUC there is no way to "share" the `backend` argument. It made sense to use it for backend-specific commands, but I don't see a way to use it for the "all-in-one" command here.

Reply:
no we don't need to provide `cloud_provider` see `create_runner_instance`, it can read the provider from click's context.

Reply:
Ok, that makes sense. I've added `CloudRegion`.